src/Microsoft.PowerShell.Commands.Management/commands/management/Navigation.cs

```diff
@@ -3519,22 +3519,87 @@ protected override bool ProviderSupportsShouldProcess
 
         #region Command code
 
+        private Collection<PathInfo> GetResolvedPaths(string path)
+        {
+            Collection<PathInfo> results = null;
+            try
+            {
+                results = SessionState.Path.GetResolvedPSPathFromPSPath(path, CmdletProviderContext);
+            }
+            catch (PSNotSupportedException notSupported)
+            {
+                WriteError(
+                    new ErrorRecord(
+                        notSupported.ErrorRecord,
+                        notSupported));
+            }
+            catch (DriveNotFoundException driveNotFound)
+            {
+                WriteError(
+                    new ErrorRecord(
+                        driveNotFound.ErrorRecord,
+                        driveNotFound));
+            }
+            catch (ProviderNotFoundException providerNotFound)
+            {
+                WriteError(
+                    new ErrorRecord(
+                        providerNotFound.ErrorRecord,
+                        providerNotFound));
+            }
+            catch (ItemNotFoundException pathNotFound)
+            {
+                WriteError(
+                    new ErrorRecord(
+                        pathNotFound.ErrorRecord,
+                        pathNotFound));
+            }
+
+            return results;
+        }
+
         /// <summary>
         /// Moves the specified item to the specified destination
         /// </summary>
         protected override void ProcessRecord()
+        {
+            if (SuppressWildcardExpansion)
+            {
+                RenameItem(Path, literalPath: true);
+                return;
+            }
+
+            Collection<PathInfo> resolvedPaths = GetResolvedPaths(Path);
+
+            if (resolvedPaths == null)
+            {
+                return;
+            }
+
+            if (resolvedPaths.Count == 1)
+            {
+                RenameItem(resolvedPaths[0].Path, literalPath: true);
+            }
+            else
+            {
+                RenameItem(WildcardPattern.Unescape(Path), literalPath: true);
+            }
+        }
+
+        private void RenameItem(string path, bool literalPath = false)
         {
             CmdletProviderContext currentContext = CmdletProviderContext;
+            currentContext.SuppressWildcardExpansion = literalPath;
```

Comment:
seems like `literalPath` doesn't need to be passed around as you can just use `SuppressWildcardExpansion` from base?

Reply:
I believe it would be unable to resolve `-Path` as literal path if we use the `base.SuppressWildcardExpansion` here.